bin/src/hickory-dns.rs

```diff
@@ -654,6 +688,115 @@ where
     }
 }
 
+#[cfg(feature = "metrics")]
+struct ConfigMetrics {
+    #[cfg(feature = "resolver")]
+    zones_forwarder: Counter,
+
+    zones_primary_file: Counter,
+    zones_secondary_file: Counter,
+    #[cfg(feature = "sqlite")]
+    zones_primary_sqlite: Counter,
+    #[cfg(feature = "sqlite")]
+    zones_secondary_sqlite: Counter,
+}
+
+#[cfg(feature = "metrics")]
+impl ConfigMetrics {
+    fn new(config: &Config) -> Self {
+        let hickory_info = gauge!("hickory_info", "version" => hickory_client::version());
+        describe_gauge!("hickory_info", Unit::Count, "hickory service metadata");
+        hickory_info.set(1);
+
+        let hickory_config_info = gauge!("hickory_config_info",
+            "directory" => config.directory().to_string_lossy().to_string(),
+            "disable_https" => config.disable_https().to_string(),
+            "disable_quic" => config.disable_quic().to_string(),
+            "disable_tcp" => config.disable_tcp().to_string(),
+            "disable_tls" => config.disable_tls().to_string(),
+            "disable_udp" => config.disable_udp().to_string(),
+            "allow_networks" => config.allow_networks().len().to_string(),
+            "deny_networks" => config.deny_networks().len().to_string(),
+            "zones" => config.zones().len().to_string()
+        );
+        describe_gauge!(
+            "hickory_config_info",
+            Unit::Count,
+            "hickory config metadata"
+        );
+        hickory_config_info.set(1);
+
+        let zones_total_name = "hickory_zones_total";
+        let zones_primary_file = counter!(zones_total_name, "store" => "file", "role" => "primary");
+        let zones_secondary_file =
```

Comment:
For these names, it feels like `file_primary` would be more intuitive hierarchically instead of `primary_file`? Same for `sqlite` etc.

Reply:
Updated the names.